builtin/logical/pki/path_root.go

```diff
@@ -451,9 +452,12 @@ func (b *backend) pathCASignSelfIssued(ctx context.Context, req *logical.Request
 	}
 
 	if signingPubType != certPubType {
-		b, ok := data.GetOk("allow_different_signature_algorithm")
-		if ok && b.(bool) {
+		b, ok := data.GetOk("require_matching_certificate_algorithms")
+		if !ok || !b.(bool) {
```

Comment:
Check looks correct though! :) 

 - Err: assume default of False; succeed
 - False value: Use the signature algorithm.
 - True value: Drop to the err case below.